tests/object/stage/test_stage.py

```diff
@@ -233,6 +253,100 @@ def test_copy_throws_error_for_same_platform_operation(runner, source, dest, sna
     assert result.output == snapshot
 
 
+@pytest.mark.parametrize(
+    "stage_path, files_on_stage, expected_calls",
+    [
+        (
+            "@exe",
+            ["a/s2.sql", "a/b/s3.sql", "s1.sql"],
+            [
+                "get @exe/a/s2.sql file://{}/a/ parallel=4",
+                "get @exe/a/b/s3.sql file://{}/a/b/ parallel=4",
+                "get @exe/s1.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/",
+            ["a/s2.sql", "a/b/s3.sql", "s1.sql"],
+            [
+                "get @exe/a/s2.sql file://{}/a/ parallel=4",
+                "get @exe/a/b/s3.sql file://{}/a/b/ parallel=4",
+                "get @exe/s1.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "snow://exe",
+            ["a/s2.sql", "a/b/s3.sql", "s1.sql"],
+            [
+                "get @exe/a/s2.sql file://{}/a/ parallel=4",
+                "get @exe/a/b/s3.sql file://{}/a/b/ parallel=4",
+                "get @exe/s1.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/a",
+            ["a/s2.sql", "a/b/s3.sql"],
+            [
+                "get @exe/a/s2.sql file://{}/ parallel=4",
+                "get @exe/a/b/s3.sql file://{}/b/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/a/",
+            ["a/s2.sql", "a/b/s3.sql"],
+            [
+                "get @exe/a/s2.sql file://{}/ parallel=4",
+                "get @exe/a/b/s3.sql file://{}/b/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/a/b",
+            ["a/b/s3.sql"],
+            [
+                "get @exe/a/b/s3.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/a/b/",
+            ["a/b/s3.sql"],
+            [
+                "get @exe/a/b/s3.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/a/b/s3.sql",
+            ["a/b/s3.sql"],
+            [
+                "get @exe/a/b/s3.sql file://{}/ parallel=4",
+            ],
+        ),
+        (
+            "@exe/s1.sql",
+            ["s1.sql"],
+            [
+                "get @exe/s1.sql file://{}/ parallel=4",
+            ],
+        ),
+    ],
+)
+@mock.patch(f"{STAGE_MANAGER}._execute_query")
+def test_copy_directory_structure(
+    mock_execute, mock_cursor, temp_dir, stage_path, files_on_stage, expected_calls
+):
+    mock_execute.return_value = mock_cursor(
+        [{"name": f"exe/{file}"} for file in files_on_stage], []
+    )
+
+    StageManager().get(stage_path, Path(temp_dir))
+
+    assert mock_execute.mock_calls[0] == mock.call(
```

Comment:
Consider defining `ls_call, *get_calls =  mock_execute.mock_calls` for better readability

Reply:
Looks much better